### ISSUE TITLE
False positive for OutputNotEscaped in ternary with braces

### ISSUE THREAD

**Issue by MatzeKitt** [WordPress/WordPress-Coding-Standards#2006]
## Bug Description
If I use optional braces in my ternary, I get a false positive error for `WordPress.Security.EscapeOutput.OutputNotEscaped`.

## Minimal Code Snippet
Not okay:
```php
echo ( $attributes[ $attribute ] ? ' ' . esc_html( $attribute ) . '="' . esc_attr( $attributes[ $attribute ] ) . '"' : '' );
```
Or:
```php
echo ' data-validate-minmax="' . esc_attr( $attributes['min'] ) . ( $attributes['max'] ? ',' . esc_attr( $attributes['max'] ) : '' ) . '"';
```

Okay:
```php
echo $attributes[ $attribute ] ? ' ' . esc_html( $attribute ) . '="' . esc_attr( $attributes[ $attribute ] ) . '"' : '';
```

The issue happens when running this command:
```bash
$ phpcs -s --ignore=assets/js/validator.*,assets/js/flatpickr.*,assets/js/i10n/*,*src/*,*build/* *

FILE: /file.php
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINES
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 2 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$attributes[$attribute]'.
     |       | (WordPress.Security.EscapeOutput.OutputNotEscaped)
```

... over a file containing this code:
```php
<?php
echo ( $attributes[ $attribute ] ? ' ' . esc_html( $attribute ) . '="' . esc_attr( $attributes[ $attribute ] ) . '"' : '' );
```

The same applies to this code:
```php
<?php
echo ' data-validate-minmax="' . esc_attr( $attributes['min'] ) . ( $attributes['max'] ? ',' . esc_attr( $attributes['max'] ) : '' ) . '"';
```

## Error Code
WordPress.Security.EscapeOutput.OutputNotEscaped

## Environment

| Question               | Answer
| ------------------------| -------
| PHP version             | 7.3.29-1+0~20210701.86+debian9~1.gbp7ad6eb
| PHP_CodeSniffer version | 3.6.0
| WPCS version            | 2.3.0
| WPCS install type       | Composer global

## Tested Against `develop` branch?
- [ ] I have verified the issue still exists in the `develop` branch of WPCS.

**Comment by jrfnl** [comment]
Duplicate of #677 and #1219

**Comment by jrfnl** [comment]
FYI: this will be fixed in WPCS 3.0.0.
